compiler/rustc_codegen_ssa/src/mir/operand.rs

```diff
@@ -518,6 +518,10 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         match self.locals[place_ref.local] {
             LocalRef::Operand(mut o) => {
                 // Moves out of scalar and scalar pair fields are trivial.
+                o.layout.ty = self.monomorphized_place_ty(mir::PlaceRef {
+                    local: place_ref.local,
+                    projection: &[],
+                });
```

Comment:
unsure whether this one is necessary, should maybe move this to an assert when setting a `LocalRef` :thinking: